core/src/main/java/com/cloudinary/android/AndroidJobStrategy.java

```diff
@@ -2,273 +2,173 @@
+        Operation operation = WorkManager.getInstance().cancelAllWorkByTag(requestId);
+        return operation.getResult().isCancelled();
     }
 
-    private void killThread(String requestId) {
-        synchronized (threadsMapLockObject) {
-            WeakReference<Thread> ref = threads.remove(requestId);
-            if (ref != null) {
-                Thread thread = ref.get();
-                if (thread != null) {
-                    thread.interrupt();
-                }
-
-                ref.clear();
-            }
-        }
-    }
-
-    /**
-     * {@inheritDoc}
-     */
     @Override
     public int cancelAllRequests() {
-        Logger.i(TAG, "All requests cancelled.");
-        int count = JobManager.instance().cancelAll();
-        killAllThreads();
-        return count;
+        WorkManager.getInstance().cancelAllWork();
+        return 0;
     }
 
-    private void killAllThreads() {
-        synchronized (threadsMapLockObject) {
-            for (String requestId : threads.keySet()) {
-                WeakReference<Thread> ref = threads.get(requestId);
-                Thread thread = ref.get();
-
-                if (thread != null) {
-                    thread.interrupt();
-                }
-
-                ref.clear();
-            }
-
-            threads.clear();
-        }
-    }
-
-    /**
-     * {@inheritDoc}
-     */
     @Override
     public int getPendingImmediateJobsCount() {
-        int pending = 0;
-        for (JobRequest jobRequest : JobManager.instance().getAllJobRequests()) {
-            if (isImmediate(jobRequest)) {
-                pending++;
-            }
-        }
-
-        return pending;
+        return getJobCountByState(WorkInfo.State.ENQUEUED);
     }
 
-    private boolean isImmediate(JobRequest jobRequest) {
-        return jobRequest.getStartMs() < IMMEDIATE_THRESHOLD;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
     @Override
     public int getRunningJobsCount() {
-        int running = 0;
-        for (Job job : JobManager.instance().getAllJobs()) {
-            if (!job.isFinished()) {
-                running++;
-            }
-        }
-
-        return running;
+        return getJobCountByState(WorkInfo.State.RUNNING);
     }
 
-    private static class JobCreator implements com.evernote.android.job.JobCreator {
-
-        @Override
-        public Job create(String tag) {
-            return new UploadJob();
+    private int getJobCountByState(WorkInfo.State state) {
+        int counter = 0;
+        List<WorkInfo> list = WorkManager.getInstance().getWorkInfosByTagLiveData(JOB_TAG).getValue();
+        if (list != null) {
+            for (WorkInfo info : list) {
+                if (info.getState() == state) {
+                    counter++;
+                }
+            }
+            return counter;
         }
+        return 0;
     }
 
-    private static final class UploadJob extends Job {
+    public static final class UploadJob extends Worker {
```

Comment:
what is the reason for making it public?

Reply:
Since it extends Worker class (which is public abstract) it needs to public